### PR TITLE
Use github.com/juju/mgo/v2 and convert to go mod

### PR DESCRIPTION
**Pull request by wallyworld**
Use github.com/juju/mgo/v2 and convert to go mod

### CODE REVIEW

File: incrementalprune_test.go [hunk 1]
```diff
@@ -152,7 +152,7 @@ func (s *IncrementalPruneSuite) TestPruneCleansUpStash(c *gc.C) {
 	txnsStash := s.db.C("txns.stash")
 	stashId := bson.D{{"c", "docs"}, {"id", "1"}}
 	c.Assert(txnsStash.FindId(stashId).One(&doc), jc.ErrorIsNil)
-	// Old versions of gopkg.in/mgo.v2 di not clean up the txn-queue during remove
+	// Old versions of github.com/juju/mgo/v2 di not clean up the txn-queue during remove
```
Comment (tlm) [inline incrementalprune_test.go:155]: did*

File: incrementalprune_test.go [hunk 2]
```diff
@@ -246,7 +246,7 @@ func (s *IncrementalPruneSuite) TestPruneLeavesIncompleteStashAlone(c *gc.C) {
...
```
Comment (tlm) [inline incrementalprune_test.go:249]: did*

**Review by tlm — APPROVED** [review]
lgtm

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**